helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/waged/WagedResourceAccessor.java

```diff
@@ -0,0 +1,107 @@
+package org.apache.helix.rest.server.resources.helix.waged;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Response;
+
+import org.apache.helix.HelixAdmin;
+import org.apache.helix.HelixException;
+import org.apache.helix.ZNRecord;
+import org.apache.helix.model.IdealState;
+import org.apache.helix.model.ResourceConfig;
+import org.apache.helix.rest.server.resources.helix.AbstractHelixResource;
+import org.apache.helix.rest.server.resources.helix.ResourceAccessor;
+import org.codehaus.jackson.type.TypeReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+@Path("/clusters/{clusterId}/waged/resources")
```

Comment:
@dasahcc 
That was my initial thought, but considering there could be various types of resources, I didn't think putting a boolean flag was a scalable solution in case we might have different modes of adding resources.

For example, we may have just full-auto and waged, but in the future, we might want to have a different version of Waged rebalancer (that might use a different algorithm for example), so I just decided to create a new path where appropriate.

So in the future, we could have 

/{clusterId}/waged/resources
/{clusterId}/customWaged/resources
/{clusterId}/fastwaged/resources
...

What do you think?

Reply:
From REST path hierarchy perspective, this structure does not look good to me. Since when we design the path, the abstract of types will be the prefix path and the specific component will be the next one. For example, cluster/{clusterId}. instance/{InstanceName}.

In this case, it should be /assignmentAlgorithm/{assignmentAlgorithmType}. But resource should be the higher level since resource conceptually is the component under cluster. Also waged, customWaged or fastWaged are all different rebalance type, which is part of resource property and similar to custom, semi-auto, full-auto rebalancer. I would suggest use regular resource path instead of creating new one.

Reply:
@dasahcc 
This is a valid comment. I think perhaps another way that conforms to the REST idiom better is to just provide it as a command (QueryParam) in the existing ResourceAccessor. This way, 
1. we can keep using the existing REST endpoint, 
2. users can specify in the command how they want to add the resource,
3. the logic allows for more modes of addition (we could support new commands).

What do you think? I think I could proceed in this direction. I could apply a similar principle to GET verbs for validation.

Reply:
Great! Thanks for taking care of it.